pyproject.toml

```diff
@@ -1,2 +1,192 @@
 [tool.black]
-target-version = ['py311']
+target-version = ['py312']
+
+# This pyproject.toml is setup so it can be used with or without poetry and also
+# supports editable installs (PEP 660) without breaking IDE and linter inspection.
+
+# DEVELOPER INSTALL INSTRUCTIONS (assuming current directory is at the top of the project) :
+# Poetry:
+#   poetry install --extras dev
+# PIP:
+#   pip install -e .[dev]
+
+extend-exclude = '''
+# A regex preceded with ^/ will apply only to files and directories
+# in the root of the project.
+(
+    \.idea
+    | \.git
+    | \.mypy_cache
+    | \.tox
+    | \/docs
+    | ^/setup.py
+)
+
+'''
+
+[tool.poetry]
+name = "screenpy_playwright"
+version = "0.0.1"
+description = "ScreenPy extension to enable interacting with Playwright."
+authors = ["Perry Goy <perry.goy@gmail.com>"]
+maintainers = ["Marcel Wilson"]
+license = "MIT"
+repository = "https://github.com/ScreenPyHQ/screenpy_playwright"
+documentation = "https://screenpy-playwright-docs.readthedocs.io"
+readme = "README.md"
+classifiers = [
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Quality Assurance",
+    "Topic :: Software Development :: Testing",
+    "Topic :: Software Development :: Testing :: BDD",
+    "Natural Language :: English",
+    "License :: OSI Approved :: MIT License",
+]
+
+
+# It's possible to add optional dependencies with the poetry CLI tool using:
+#   poetry add --optional some_dev_pkg
+# This will set `optional = true` flag in [tool.poetry.dependencies], as seen below
+# But then you need to remember to manually edit the [tool.poetry.extras] dev section
+# and declare the package. This allows `pip install .[dev]` to work as expected
+# Poetry users will need to use the `--extras dev` option rather than the `--with dev`
+# so we dont have two different sets of package versions to update.
+
+[tool.poetry.dependencies]
+python = "^3.8"
+screenpy = ">=4.0.2"
```

Comment:
It needs at least this level of ScreenPy, since that's when ScreenPy busted out into the core+extensions setup we have now. I don't know if there's a reason to require a higher level of ScreenPy.